### DISCUSSION TITLE
No `dir` property in UserConfigExport

### DISCUSSION THREAD

**Discussion by crs1138** [vitest-dev/vitest#6790] [Q&A] [answered]
I have a monorepo with defined Vitest workspaces. In the configuration of a workspace, I define the `root` property pointing to the root of the monorepo. In the documentation, there's a mention of the `dir` property for the config – https://vitest.dev/config/#dir. However, Typescript tells me that this property does not exist in the configuration object: _Object literal may only specify known properties, and 'dir' does not exist in type 'UserConfigExport'._

What am I missing?

**Comment by hi-ogawa** [comment, marked as answer]
While `root` is Vite option, `dir` is Vitest option, so it should look like this
https://stackblitz.com/edit/vitest-dev-vitest-wq8au2?file=vite.config.ts

```ts
export default defineConfig({
  // [not here]
  // dir: "src"

  test: {
    // [but here]
    dir: 'src',
  },
});
```

**Comment by sheremet-va** [comment]
The documentation page has a warning at the start:

<img width="718" alt="Screenshot 2024-10-25 at 16 42 05" src="https://github.com/user-attachments/assets/f6270c24-3c0f-4d1c-a2a0-decf1fe7972d">

**Reply by crs1138** [reply]
Doh! You're right. It's just a lot of information doing the migration and I landed in the middle of the page when doing a search, thus completely missed the warning. Still, no excuse. Thanks for your response.
